### PR TITLE
fixing cross entropy loss function

### PR DESCRIPTION
**Pull request by Tizpaz**
## Description

The pull request is to fix the bug in the loss function implementation of cross entropy, as pointed out by @SeanMcCarren in the issue #1209. 

## Tests

- [ ] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted
I used the program written by the bug reporter. It now works on this program. Maybe more tests need to run, so I will check it with others to see if we need to add new tests or adjust existing ones. 

## Documentation

- [X] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Comment by SeanMcCarren** [comment]
LGTM :)

**Review by romanlutz — APPROVED** [review]

**Review by hildeweerts — APPROVED** [review]
@fairlearn/fairlearn-maintainers feel free to merge once the checks are complete

**Comment by hildeweerts** [comment]
Hi @Tizpaz. My apologies for taking so long to respond to your PR. Some of the [tests are failing](https://github.com/fairlearn/fairlearn/actions/runs/4690668586/jobs/8314092589?pr=1226#step:6:209), I think you need to update [the tests](https://github.com/fairlearn/fairlearn/blob/bc122366afecf3adacde3ebbf75815c92c2c4081/test/unit/adversarial/helper.py#L48) to reflect the change from `NLLLoss` to `CrossEntropyLoss`.

**Comment by Tizpaz** [comment]
Thanks @hildeweerts for the follow-up. I will look into this to update the test cases, Thanks!
